cookbooks/bcpc-hadoop/recipes/configs.rb

```diff
@@ -62,9 +64,69 @@
 # Install Java
 include_recipe "bcpc-hadoop::java_config"
 include_recipe "java::default"
+include_recipe "java::oracle_jce"
 
 %w{zookeeper}.each do |pkg|
   package pkg do
     action :upgrade
   end
 end
+
+# Create Keytabs (if kerberos is eanbled)
+if node[:bcpc][:hadoop][:kerberos][:enable] == true then
+
+  keytab_dir = node[:bcpc][:hadoop][:kerberos][:keytab][:dir]
+  
+  # Create directory to store keytab files
+  directory "#{keytab_dir}" do
+    owner "root"
+    group "root"
+    recursive true
+    mode 0755  
+  end
+
+
+  # Download and create keytab file
+  node[:bcpc][:hadoop][:kerberos][:data].each do |srvc, srvdat|
+  
+    keytab_file = srvdat['keytab']
+
+    config_host = srvdat['princhost'] == "_HOST" ?  float_host(node[:hostname]) : srvdat['princhost'].split('.')[0]
+    
+    # Delete existing keytab if keytab is being re-created
+    file "#{keytab_dir}/#{keytab_file}" do
+      action :delete
+      only_if { File.exists?("#{keytab_dir}/#{keytab_file}") && 
+                node[:bcpc][:hadoop][:kerberos][:keytab][:recreate] == true
+              }
+    end 
+
+    # Create the keytab file
+    file "#{keytab_dir}/#{keytab_file}" do
+      owner "#{srvdat['owner']}"
+      group "root"
+      mode "#{srvdat['perms']}"
+      action :create_if_missing
+      content Base64.decode64(get_config!("#{config_host}-#{srvc}"))
+      only_if { user_exists?("#{srvdat['owner']}")  }
+    end
+
+    princ_host = srvdat['princhost'] == "_HOST" ? float_host(node[:fqdn]) : srvdat['princhost']
+
+    next if srvdat['principal'] == "HTTP"
+
+    execute "kdestroy-for-#{srvdat['owner']}" do
```

Comment:
Do we need to `kdestroy`? The `kinit` will do a `kdestroy` if necessary as I understand it.